grails-app/services/org/pih/warehouse/picklist/PicklistService.groovy

```diff
@@ -322,4 +324,32 @@ class PicklistService {
             order("lastUpdated", "desc")
         }
     }
+
+    void revertPick(String id, String itemId) {
+        Picklist picklist = Picklist.get(id)
+        RequisitionItem requisitionItem = RequisitionItem.get(itemId)
+        List<PicklistItem> picklistItemsToRemove = PicklistItem.findAllByRequisitionItem(requisitionItem)
+
+        if (!picklist) {
+            throw new ObjectNotFoundException(id, Picklist.class.toString())
+        }
+
+        if (!picklistItemsToRemove?.size()) {
+            throw new ObjectNotFoundException(itemId, PicklistItem.class.toString())
+        }
+
+        List<String> binLocations = []
+        picklistItemsToRemove.each {
+            picklist.removeFromPicklistItems(it)
+            binLocations.add(it.binLocation?.id)
+            it.delete(flush: true)
+        }
+
+        productAvailabilityService.refreshProductsAvailability(
+                picklist?.requisition?.origin?.id,
+                picklistItemsToRemove?.requisitionItem?.product?.id?.unique(),
```

Comment:
If you have `requisitionItem`, then having `[requisitionItem?.product?.id]` is enough, because there is no way there will be different products for picklist items for the same requisition item.